uploadfiles/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+# ##############################################################################
+# apps/uploadfiles/CMakeLists.txt
```

Comment:
should we move uploadfiles to examples or system folder?

Reply:
> should we move uploadfiles to examples or system folder?

True.  It doesn't warrant residing at the top level in the directory tree.  The top level is reserved for groupings of multiple similar apps and/or libraries.  examples/ really should just contain instructional apps; system was originally intended to hold other NSH add-on  commands.

fsutils is another possibility.

Reply:
@xiaoxiang781216 my idea on putting it at root of apps/ is to let people discover it easily.

Reply:
if so, different people want to prompt the different apps to the top level.  BTW, it's simpler to utilize etc romfs to achieve the functionality.

Reply:
@xiaoxiang781216 but the romfs currently supported on NuttX needs many steps. The approach here is to create a simple a folder/directory where the user just drop the files to be included as a romfs in the flash.

I created it because I wanted to use a wave file to test on nxplayer and everytime I needed to transfer it over zmodem.

Should it be better to have this romfs solution into nuttx/ instead of apps/ ?

Reply:
> @xiaoxiang781216 my idea on putting it at root of apps/ is to let people discover it easily.

But what is so special about this app?  Couldn't any app make that same argument?  I think it is not right to put one app at the top level (excepting NSH).  It is better to keep the directory structure clean.  Can you imagine what would happen if all app developers wanted there own apps to easiest to find.  That is kind of a marketing ploy and not a reasonable decision.

Reply:
There is not special with the application, actually the application could be anyware inside the apps/[examples,system,...]. We just need a directory at apps/ ie: "apps/putyourfileheretobeincludedintonuttxnsh" to let users know that can copy the files to there!

Reply:
> #2256 already move the most part to nuttx side, the process is simple to drop the file into etc folder and add the file to RAWSRCS in Makefile.

Nice @xiaoxiang781216 it makes thing easier, so I don't need to create it here. May idea was to create something like "nuttx/userdata/" to let user put files there and be integrated/mounted automatically at /data in NSH. Just to make file easier to end users.

Putting musics and user files at /etc doesn't seems correct, what do you think? Make we could have a CONFIG_USERDATA to be similar (and share code) to CONFIG_ETC_ROMFS. Makes sense for you?

Reply:
Yes, this is something we can extend.

Reply:
Ok, I will close this PR and will submit the patch to the kernel using the ETC_ROMFS approach